doc/SONiC_Port_Mirroring_HLD.md

```diff
@@ -0,0 +1,319 @@
+      * [1.4 Warm Boot Requirements](#14-warm-boot-requirements)
+  * [2. Functionality](#2-functionality)
+      * [2.1 Functional Description](#21-functional-description)
+  * [3. Design](#3-design)
+      * [3.1 Overview](#31-overview)
+      * [3.2 DB Changes](#32-db-changes)
+          * [3.2.1 CONFIG DB](#321-config-db)
+          * [3.2.2 APP_DB](#322-app_db)
+          * [3.2.3 STATE_DB](#323-state_db)
+          * [3.2.4 ASIC_DB](#324-asic_db)
+          * [3.2.5 COUNTER_DB](#325-counter_db)
+      * [3.3 Switch State Service Design](#33-switch-state-service-design)
+          * [3.3.1 Orchestration Agent](#331-orchestration-agent)
+          * [3.3.2 Other Process](#332-other-process)
+      * [3.4 SAI](#35-sai)
+      * [3.5 CLI](#36-cli)
+          * [3.5.1 Data Models](#351-data-models)
+          * [3.5.2 Configuration Commands](#352-configuration-commands)
+          * [3.5.3 Show Commands](#353-show-commands)
+          * [3.5.4 Clear Commands](#354-clear-commands)
+          * [3.5.5 Debug Commands](#355-debug-commands)
+          * [3.5.6 Rest API Support](#356-rest-api-support)
+          * [3.5.7 GNMI Support](#357-gnmi-support)
+  * [4. Flow Diagrams](#4-flow-diagrams)
+  * [5. Error Handling](#5-Error-Handling)
+  * [6. Serviceability and Debug](#6-serviceability-and-debug)
+  * [7. Warm Boot Support](#7-warm-boot-support)
+  * [8. Scalability](#8-scalability)
+  * [9. Unit Test](#9-unit-test)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                         |
+|:---:|:-----------:|:------------------:|--------------------------------------------|
+| 0.1 | 05/17/2019  |   Rupesh Kumar      | Initial version                            |
+
+
+# About this Manual
+This document provides general information about extending mirroring implementation in SONiC.
+# Scope
+This document describes the high level design of Mirroring Enhancements feature. 
+
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+|   SPAN                   |  Switched Port ANalyzer                |
+|   ERSPAN                 |  Encapsulated Remote Switched Port ANalyzer                |
+
+
+# 1 Requirement Overview
+## 1.1 Functional Requirements
+
+1. Port/Port-channel mirroring support
+     - Add support to mirror ingress traffic on port/port-channel to SPAN/ERPSAN mirror session.
+     - Add support to mirror egress traffic on port/port-channel to SPAN/ERSPAN mirror session.
+     - Add support to mirror both ingress/egress traffic on port/port-channel to SPAN/ERSPAN mirror session.
+
+2. Dynamic session management
+    - Allow multiple source to single destination.
+    - Each session supports mirroring from single port to single destination port.
```

Comment:
do we allow one mirror session from multiple source port to single dest port?

Reply:
Yes. we allow one mirror session from multiple source port to single destination port.